testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/TestListener.java

```diff
@@ -58,5 +93,62 @@ public void testFinished(Description description) throws Exception {
         } else {
             System.out.println("Test finished but did not start: " + fullTestName);
         }
+
+        String phase = readFile(MainAgent.tmpfile);
+
+        if(phase.startsWith("4")) {
+            String polluter = phase.split(" ")[1];
```

Comment:
Let's simplify it such that there is ever only one test target name we capture relative to, so no more of this additional polluter information.